spec/models/folder_spec.rb

```diff
@@ -20,4 +20,57 @@
       expect(three.ancestors).to eq([root, one, two])
     end
   end
+
+  describe "#all_uploads" do
+    describe "returns a list of all uploads" do
+      it "for the user if the root folder" do
+        user  = create(:user)
+        root  = create(:folder, user: user)
+        one   = create(:folder, user: user, parent: root)
+        file1 = create(:upload, folder: root)
+        file2 = create(:upload, folder: one)
+
+        expect(root.all_uploads).to eq([file1, file2])
+      end
+
+      it "in itself" do
+        user  = create(:user)
+        root  = create(:folder, user: user)
+        one   = create(:folder, user: user, parent: root)
+        file1 = create(:upload, folder: root)
+        file2 = create(:upload, folder: one)
+
+        expect(one.all_uploads).to eq([file2])
+        expect(one.all_uploads).to_not eq([file1])
+      end
+
+      it "in itself and a subfolder" do
+        user  = create(:user)
+        root  = create(:folder, user: user)
+        one   = create(:folder, user: user, parent: root)
+        two   = create(:folder, user: user, parent: one)
+        file1 = create(:upload, folder: root)
+        file2 = create(:upload, folder: one)
+        file3 = create(:upload, folder: two)
```

Comment:
it's really helpful seeing this factory pattern, sometimes it's hard to figure out how to generate them based off of their relationships - great work!